src/Features/Core/Portable/NavigateTo/IRemoteNavigateToSearchService.cs

```diff
@@ -9,7 +9,10 @@ namespace Microsoft.CodeAnalysis.NavigateTo
 {
     internal interface IRemoteNavigateToSearchService
     {
-        Task<IList<SerializableNavigateToSearchResult>> SearchDocumentAsync(DocumentId documentId, string searchPattern, string[] kinds, CancellationToken cancellationToken);
-        Task<IList<SerializableNavigateToSearchResult>> SearchProjectAsync(ProjectId projectId, DocumentId[] priorityDocumentIds, string searchPattern, string[] kinds, CancellationToken cancellationToken);
+        Task<IList<SerializableNavigateToSearchResult>> SearchDocumentAsync(
+            PinnedSolutionInfo solutionInfo, DocumentId documentId, string searchPattern, string[] kinds, CancellationToken cancellationToken);
+
+        Task<IList<SerializableNavigateToSearchResult>> SearchProjectAsync(
+            PinnedSolutionInfo solutionInfo, ProjectId projectId, DocumentId[] priorityDocumentIds, string searchPattern, string[] kinds, CancellationToken cancellationToken);
```

Comment:
is hte general idea that we want these remote calls to be more explicit about the solution snapshot they should be running against?

Reply:
Kind of. Rather, it's not to rely on any state.